internal/mode/static/nginx/config/servers.go

```diff
@@ -247,6 +247,9 @@ func updateLocationsForFilters(
 
 	rewrites := createRewritesValForRewriteFilter(filters.RequestURLRewrite, path)
 	proxySetHeaders := generateProxySetHeaders(&matchRule.Filters)
+	addResponseHeaders := generateAddResponseHeaders(&matchRule.Filters)
```

Comment:
Another bonus of introducing a `ResponseHeaders` struct is we can collapse these three lines into one:

```
responseHeaders := generateResponseHeaders(&matchRule.Filters)
```

Plus, then we only have to handle this case `if filters == nil || filters.ResponseHeaderModifiers == nil {
		return []http.Header{}
	}` once.

Reply:
Update https://github.com/nginxinc/nginx-gateway-fabric/commit/629ee127585f277e3e41eafc1d47fd86ea583c43 https://github.com/nginxinc/nginx-gateway-fabric/pull/1494/commits/d8b35c0ceeb14c45b416469dc6e7480a04ff1329